integr8ly/tutorial-web-app
# INTLY-1362 Use managedServices in walkthrough manifest

Currently, we provision "slices" of AMQ Online and Fuse Online when
a user logs in for the first time. However, a user may never use a
provisioned "slice", meaning resources are being wasted on a cluster.

This change allows these "slices" to be defined in the Walkthrough
configuration itself, so that they will be provisioned as-neeeded.

This should help with resource usages in the cluster.

Verification:
- Run the server, pointing at an Integreatly cluster from `master`
and with `WALKTHROUGH_LOCATIONS` pointing to
`https://github.com/aidenkeating/tutorial-web-app-walkthroughs.git#INTLY-1362-add-shared-services-config`
- [x] Ensure that on first login, AMQ Online and the non-Shared Fuse
Online are no longer being provisioned (check the namespace).
- [x] Ensure every service except AMQ Online on the right-hand side is clickable and the dashboards show as expected
- [x] Ensure that AMQ Online is visible in the right-hand bar of the landing page with a `Launch` button.
- [x] Click the `Launch` button, ensure the button disappears and the service starts provisioning
- [x] Ensure AMQ Online eventually hits a provisioned state in the landing page
- Click on Walkthrough 1A
- [x] Ensure the loading screen has the same layout as the initial loading screen (see images in comments).
- [x] Ensure the loading screen stays visible
until the AMQ Online and Fuse Onlne slices have been fully provisioned
and the NodeJS Application is provided with the correct credentials.
- Load another Walkthrough, ensure they still load as expected without issues.
